static/docs/commands-reference/repro.md

```diff
@@ -15,25 +15,83 @@ name is omitted, `Dvcfile` will be used by default.
+pipeline is defined using the `dvc run` command.
+
+There are several ways to limit the stages to rerun:
+
+* Listing stage file(s) as `targets`
+* The `--single-item` option ensures that only a single stage is rerun.
+* The `--pipeline` option limits execution to the pipeline
+ containing the named stage, especially useful if the
+ workspace contains multiple pipelines.  
+* The `--cwd` option limits execution to the pipeline 
+ located in a subdirectory.
+
+There are two ways to ensure rerunning additional stages beyond
+a changed stage:
+
+* The `--force` option causes the entire pipeline to be rerun
+* The `--ignore-build-cache` causes all stages following a changed stage
+ to be rerun.
+
 `dvc repro` does not run `dvc fetch`, `dvc pull` or `dvc checkout` to get source
 data files, intermediate or final results.
 
+## Options
+
+* `-q`, `--quiet` do not write anything to standard output.  The command
+  run by the stage is free to make output irregardless of this flag.
+  Exit with 0 if all stages are up to date or if all stages
+  are successfully rerun, otherwise exit with 1. 
+
+* `-v`, `--verbose` displays detailed tracing information from executing the
+  `dvc repro` command.
+
+* `-f`, `--force`  Rerun the pipeline, reproducing its results, even
+ if no changes were found.  By default this reruns the entire pipeline.
+ To rerun a single stage, specify the stage name on the command-line
+ along with the `--single-item` option.
+
+* `-s`, `--single-item`  Reproduce only a single data item without 
+ recursively checking for changed dependencies.
+ Multiple stages are rerun if multiple stage names are listed on the command-line.
+
+* `-c`, `--cwd`  Directory within your project to reproduce from.  If no 
+ target names are given, it attempts to use `Dvcfile`, if it exists
+ in the named directory, for stages to rerun.
+ Instead of using `--cwd` one can alternately specify a target in
+ a subdirectory as `path/to/target.dvc`.
+
+* `-m`, `--metrics`  Show metrics after reproduction.  The pipeline must
+ have at least one metrics file defined either with the `dvc metrics` command,
+ or by the `--metrics-no-cach` option on the `dvc run` command.
+
+* `--dry`  Only print the commands that would be executed without
+ actually executing the commands.
+
+* `-i`, `--interactive`  Ask for confirmation before reproducing each stage.
+ The stage is rerun if the user types "y".
+
+* `-p`, `--pipeline`  Reproduce the whole pipeline that the specified stage
+ file belongs to.  Use `dvc pipeline show target.dvc` to show the entire
+ pipeline the named stage belongs to.
+
+* `--ignore-build-cache`  Reproduce all descendants of a changed stage, or 
+ the stages following the changed stage, even if their direct dependencies
+ did not change.  Like with the same option on `dvc run`, this is a way to
+ force certain stages to run if they would not otherwise be rerun.
+
+
 ## Examples
 
 Reproduce default stage file:
```

Comment:
I think it would be better if illustrate all the command with as simple examples as possible, let's not rely on some complex pipelines from the tutorial. If we use some pipeline from other example, let's make a reference. There should be some introduction that explains what stages does it have, how these stages connected with each. Can we start by creating some very simple pipeline with two steps. Just simple text file, count lines, or something. We can even mention that this pipeline is very artificial and serves only illustrative purposes.

Reply:
I've come up with a slightly longer example.  One reason being to have an example of a stage with nondeterministic results.  So I threw in a stage using sort --sort-random to have a random sort.